static/js/compose.js

```diff
@@ -778,13 +778,39 @@ function validate_stream_message() {
     return true;
 }
 
+// The function will check whether the recipient is in settings.CROSS_REALM_BOT_EMAILS or settings.ADMIN_DOMAIN
 function validate_private_message() {
     if (exports.recipient() === "") {
         compose_error("Please specify at least one recipient", $("#private_message_recipient"));
         return false;
-    }
+    } else {
+        var private_recipients = util.extract_pm_recipients();
+        var invalid_recipients = [];
+        _.each(private_recipients, function (email) {
+            // This case occurs when exports.recipient() ends with ','
+            if (email === "") {
+                return;
+            }
+            if (util.string_in_list_case_insensitive(email, page_params.cross_realm_bot_emails)) {
+                return;
+            }
+            var index = email.length - page_params.domain.length;
+            // The email should contain '@' and atleast one char before '@'
+            if ((index <= 1) || (email[index - 1] !== '@') || (email.substring(index).toLowerCase() !== page_params.domain.toLowerCase())) {
+                invalid_recipients.push(email);
+            }
```

Comment:
Yep.  The domain is used only for realms where `restricted_to_domain=True` and only for the process of allowing new users to join the realm.